pandas/tests/indexes/interval/test_interval_tree.py

```diff
@@ -12,33 +12,36 @@ def closed(request):
     return request.param
 
 
+@pytest.fixture(
+    scope='class', params=['int32', 'int64', 'float32', 'float64', 'uint64'])
+def dtype(request):
+    return request.param
+
+
+@pytest.fixture(scope='class')
+def tree(dtype):
+    left = np.arange(5, dtype=dtype)
+    return IntervalTree(left, left + 2)
+
+
 class TestIntervalTree(object):
     def setup_method(self, method):
-        def gentree(dtype):
-            left = np.arange(5, dtype=dtype)
-            right = left + 2
-            return IntervalTree(left, right)
-
-        self.tree = gentree('int64')
-        self.trees = {dtype: gentree(dtype)
-                      for dtype in ['int32', 'int64', 'float32', 'float64']}
-
-    def test_get_loc(self):
-        for dtype, tree in self.trees.items():
-            tm.assert_numpy_array_equal(tree.get_loc(1),
-                                        np.array([0], dtype='int64'))
-            tm.assert_numpy_array_equal(np.sort(tree.get_loc(2)),
-                                        np.array([0, 1], dtype='int64'))
-            with pytest.raises(KeyError):
-                tree.get_loc(-1)
-
-    def test_get_indexer(self):
-        for dtype, tree in self.trees.items():
-            tm.assert_numpy_array_equal(
-                tree.get_indexer(np.array([1.0, 5.5, 6.5])),
-                np.array([0, 4, -1], dtype='int64'))
-            with pytest.raises(KeyError):
-                tree.get_indexer(np.array([3.0]))
+        self.tree = tree('int64')
```

Comment:
prob don't need setup_method this if you pass tree into other methods

Reply:
The `tree` fixture is using the `dtype` fixture when passed as a parameter to tests, so it's generating trees for all dtypes in the `dtype` fixture.  Initializing `self.tree` like that is just a hack I used to create a single instance of a `int64` tree, since there was one existing test that only used `int64`.  The test passes for all dtypes though, so removed `setup_method` and just passed in the `tree` fixture.